docs/rfc/009-scalar-type-accesses-and-methods.md

```diff
@@ -0,0 +1,77 @@
+# Leo RFC 009: Scalar Type Accesses And Methods
+
+## Authors
+
+- Max Bruce
+- Collin Chin
+- Alessandro Coglio
+- Eric McCarthy
+- Jon Pavlik
+- Damir Shamanaev
+- Damon Sicore
+- Howard Wu
+
+## Status
+
+DRAFT
+
+## Summary
+
+This RFC proposes two things:
+
+1. The scalar types in Leo (integers, fields, etc.) can have static methods.
+2. The scalar types in Leo (integers, fields, etc.) can have static constants.
+3. Those values that have a scalar type can have methods directly on them.
+
+## Motivation
+
+This approach allows for a clean interface to provide built-in methods or static members for these basic types.
+
+## Design
+
+### Semantics
+
+Firstly we would have to modify both the ABNF and parsing of Leo to allow static method calls onto a scalar type.
+
+The ABNF would look as follows:
+
+```abnf
+; This is an existing old rule
+scalar-type =  boolean-type / arithmetic-type / address-type / character-type
+
+; This is an existing old rule
+circuit-type = identifier / self-type
+
+; Add this rule.
+named-type = circuit-type / scalar-type ; new rule
+
+; Modify this rule:
+postfix-expression = primary-expression
+                   / postfix-expression "." natural
+                   / postfix-expression "." identifier
+                   / identifier function-arguments
+                   / postfix-expression "." identifier function-arguments
+                   / named-type "::" identifier function-arguments ; this used to be a circuit-type
```

Comment:
For static constants, we also need to add the following alternative to this rule:
```
                   / named-type "::" identifier ; new alternative
```
This opens the door to circuit types also including member constants, but that should be probably a separate feature. (In particular, the syntax of circuit declarations should be extended for that.)

Reply:
Yea, I was thinking of a separate feature for adding those to circuits. While we are in the scope of modifying these, I just figured we should add in the constants on the scalar types at least. Then we can add completeness in another feature by allowing it on a circuit.